src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -771,47 +891,87 @@ public function next_token() {
 		 */
 		$t = $this->html[ $this->tag_name_starts_at ];
 		if (
-			! $this->is_closing_tag &&
-			(
+			$this->is_closing_tag ||
+			! (
 				'i' === $t || 'I' === $t ||
 				'n' === $t || 'N' === $t ||
 				's' === $t || 'S' === $t ||
 				't' === $t || 'T' === $t ||
 				'x' === $t || 'X' === $t
 			)
 		) {
-			$tag_name = $this->get_tag();
+			return true;
+		}
 
-			if ( 'SCRIPT' === $tag_name && ! $this->skip_script_data() ) {
-				$this->parser_state         = self::STATE_INCOMPLETE;
-				$this->bytes_already_parsed = $was_at;
+		$tag_name = $this->get_tag();
+		if (
+			'SCRIPT' !== $tag_name &&
+			'TEXTAREA' !== $tag_name &&
+			'TITLE' !== $tag_name &&
+			'IFRAME' !== $tag_name &&
+			'NOEMBED' !== $tag_name &&
+			'NOFRAMES' !== $tag_name &&
+			'STYLE' !== $tag_name &&
+			'XMP' !== $tag_name
```

Comment:
With all the alignment lints, it would actually please me to have this aligned (but the linter complained).

This works. Stylistic and up to you:

```suggestion
			// phpcs:disable WordPress.WhiteSpace.OperatorSpacing.SpacingBefore
			'SCRIPT'   !== $tag_name &&
			'TEXTAREA' !== $tag_name &&
			'TITLE'    !== $tag_name &&
			'IFRAME'   !== $tag_name &&
			'NOEMBED'  !== $tag_name &&
			'NOFRAMES' !== $tag_name &&
			'STYLE'    !== $tag_name &&
			'XMP'      !== $tag_name
			// phpcs:enable
```

Bonus, we could alphabetize, although it looks like you may have optimized this to short circuit on the tags we expect to be most common? That may be worth noting in a comment if so.

Reply:
> Stylistic and up to you:

I don't think it is up to either of us, but maybe the linting rules don't check here with ab `if` condition. in the past I've had people remove the ignore statements without asking, and revert the code to the misaligned form, so I'm a little burnt and skeptical about trying to make it more comprehensible here

Reply:
> it looks like you may have optimized this to short circuit on the tags we expect to be most common? That may be worth noting in a comment if so.

Was this the case? That comment may be worthwhile so folks give the ordering a little extra scrutiny when working with this code.

Reply:
these were organized by what content they skip over, and within those groups they were alphabetical.

in 4639ff8920189480d0f39f8c2275277d18f73f9b I've removed this entirely since we have to duplicate the check. previously it served more of a purpose when we didn't have the special-elements logic here, but now it's probably not bringing us much benefit.